File: fitcore-frontend/src/models/Workout.ts
import {WorkoutExercise} from "./WorkoutExercise";

export interface Workout {
    id: string;                // Идентификатор тренировки
    userId: string;            // Владелец тренировки
    title: string;             // Название тренировки
    date: string;              // Дата тренировки
    completed: boolean;        // Выполнена ли тренировка
    durationMinutes: number;   // Длительность в минутах
    exercises: WorkoutExercise[]; // Упражнения в тренировке
}

File: fitcore-backend/Entity/Workout.cs
namespace fitcore_backend.Entity;

/// <summary>
/// Тренировка
/// </summary>
public class Workout
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }

    public string Title { get; set; } = null!;
    public DateTime Date { get; set; } = DateTime.UtcNow;
    
    public bool Completed { get; set; } = false;
    public int DurationMinutes { get; set; }

    public User User { get; set; } = null!;
    public ICollection<WorkoutExercise> Exercises { get; set; } = new List<WorkoutExercise>();
}
